components/Article.tsx

```diff
@@ -18,17 +18,17 @@ import TocWrapper from './TocWrapper'
 
 export interface Props {
     title: string
-    author: string
+    author?: string | null
     tags: string[]
     mdxSource: MDXRemoteSerializeResult
-    image?: string
-    imageAlt?: string
-    socialImage?: string
-    description?: string
-    toc?: Node
-    collection?: RecordCollection
+    image?: string | null
+    imageAlt?: string | null
+    socialImage?: string | null
+    description?: string | null
+    toc?: Node | null
+    collection?: RecordCollection | null
```

Comment:
I'm curious about the reason for the `null` in these fields. I would assume that it's simpler to only allow `undefined`, which we already have with the optional nature of the fields, and that covers the case of the field being entirely absent from the object as well as explicitly having an `undefined` value.

Reply:
I was thinking about it for our validators — where we are basically trying to "normalize" our data.  I didn't like the idea of setting empty strings or other placeholder values that don't themselves have meaning, and may be hard to reason about in the future, and I thought `null` might be the way to go — thinking down the road as well for interfacing with a database or some such (reasoning about `NULL` versus empty strings or placeholders). If we were using PostgreSQL, for example, the default on a value that's not set would be `NULL` (unless another default is set).

Reply:
I see what you mean. Maybe if we ensure that we'll always use `null`, and not `undefined`, then we could make the fields non-optional.

Reply:
I think we should keep the fields as optional, I think the front matter can be a lot to ask for and in practice I see many contributors who work in markdown ignoring headers completely.